constructor/winexe.py

```diff
@@ -330,32 +311,71 @@ def make_nsi(
                     )
                 info[key] = valid_values
 
-    for key, value in replace.items():
+    for key, value in variables.items():
         if value.startswith('@'):
             value = join(dir_path, value[1:])
-        replace[key] = win_str_esc(value)
-
-    data = read_nsi_tmpl(info)
-    ppd = ns_platform(info['_platform'])
-    ppd['initialize_conda'] = info.get('initialize_conda', True)
-    ppd['initialize_by_default'] = info.get('initialize_by_default', None)
-    ppd['register_python'] = info.get('register_python', True)
-    ppd['register_python_default'] = info.get('register_python_default', None)
-    ppd['check_path_length'] = info.get('check_path_length', None)
-    ppd['check_path_spaces'] = info.get('check_path_spaces', True)
-    ppd['keep_pkgs'] = info.get('keep_pkgs') or False
-    ppd['pre_install_exists'] = bool(info.get('pre_install'))
-    ppd['post_install_exists'] = bool(info.get('post_install'))
-    ppd['with_conclusion_text'] = bool(conclusion_text)
-    ppd["enable_debugging"] = bool(os.environ.get("NSIS_USING_LOG_BUILD"))
-    ppd["has_conda"] = info["_has_conda"]
-    ppd["custom_welcome"] = info.get("welcome_file", "").endswith(".nsi")
-    ppd["custom_conclusion"] = info.get("conclusion_file", "").endswith(".nsi")
-    ppd["has_license"] = bool(info.get("license_file"))
-    ppd["post_install_pages"] = bool(info.get("post_install_pages"))
-
-    data = preprocess(data, ppd)
-    data = fill_template(data, replace, exceptions=nsis_predefines)
+        variables[key] = win_str_esc(value)
+
+    # From now on, the items added to variables will NOT be escaped
+
+    # These are mostly booleans we use with if-checks
```

Comment:
Now that this is all one dictionary, this could be merged into the `variables` initializer.

Reply:
We need to do it here separately because the block above have been stringified and escaped, but we want true booleans for these ones.